builder/powervs/common/ssh.go

```diff
@@ -34,9 +39,60 @@ func SSHHost() func(multistep.StateBag) (string, error) {
 			if host != "" {
 				return host, nil
 			}
+
+			dhcpServerID, ok := state.GetOk("dhcpServerID")
+			if !ok {
+				// if the dhcpServerID is not set, dont try to fetch IP from DHCP server, instead wait for address to get populated.
+				ui.Message("Machine IP is not yet found, Trying again")
+				time.Sleep(sshHostSleepDuration)
+				continue
+			}
+			dhcpClient := state.Get("dhcpClient").(*instance.IBMPIDhcpClient)
+			ui.Message("Getting Instance IP from DHCP server")
+
+			net := state.Get("network").(*models.Network)
+			networkID := net.NetworkID
+
+			var pvmNetwork *models.PVMInstanceNetwork
+			for _, network := range in.Networks {
+				if network.NetworkID == *networkID {
+					pvmNetwork = network
+					ui.Message("Found network attached to VM")
+				}
+			}
+
+			if pvmNetwork == nil {
+				ui.Message("Failed to get network attached to VM, Trying again")
+				time.Sleep(sshHostSleepDuration)
+				continue
+			}
+
+			dhcpServerDetails, err := dhcpClient.Get(dhcpServerID.(string))
+			if err != nil {
+				ui.Error(fmt.Sprintf("Failed to get DHCP server details: %v", err))
+				return "", err
+			}
+
+			if dhcpServerDetails == nil {
+				ui.Error(fmt.Sprintf("DHCP server details is nil, DHCPServerID: %s", dhcpServerID))
+				return "", err
+			}
+
+			var internalIP string
+			for _, lease := range dhcpServerDetails.Leases {
+				if *lease.InstanceMacAddress == pvmNetwork.MacAddress {
+					ui.Message(fmt.Sprintf("Found internal ip for VM from DHCP lease IP %s", *lease.InstanceIP))
+					internalIP = *lease.InstanceIP
+					break
+				}
+			}
+			if internalIP != "" {
+				return internalIP, nil
+			}
```

Comment:
Not sure whats wrong here this loop never exits.
I keep seeing this
```
    powervs: Found network attached to VM
    powervs: Found internal ip for VM from DHCP lease IP 192.168.0.10
    powervs: Fetching IP for machine
    powervs: Getting Instance IP from DHCP server
    powervs: Found network attached to VM
    powervs: Found internal ip for VM from DHCP lease IP 192.168.0.10
==> powervs: Timeout waiting for SSH.
```
@mkumatag @Prajyot-Parab any idea

Reply:
Oh , Its works as expected, until the SSH connection is established, it will keep on calling the function. Here is log of success case
```
==> powervs: Creating network
    powervs: Network Created, Name: public-192_168_230_0-29-VLAN_2000, ID: b571af15-e523-4f29-aca3-22d641cd8f02
==> powervs: Creating Instance
    powervs: Creating Instance
    powervs: Instance Created, Name: capibm-centos-streams8-1713966535, ID: 040822c1-4605-4120-ab87-e80c16b1ff6d
    powervs: Fetching IP for machine
    powervs: Machine IP is not yet found, Trying again
    powervs: Machine IP is not yet found, Trying again
==> powervs: Using SSH communicator to connect: abc.zz.yy.xx
==> powervs: Waiting for SSH to become available...
    powervs: Fetching IP for machine
    powervs: Fetching IP for machine
    powervs: Fetching IP for machine
    powervs: Fetching IP for machine
    powervs: Fetching IP for machine
    powervs: Fetching IP for machine
==> powervs: Connected to SSH!
```